File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/pool/ClosePoolTest.java
package org.hyperledger.indy.sdk.pool;

import org.hyperledger.indy.sdk.IndyIntegrationTest;
import org.hyperledger.indy.sdk.utils.PoolUtils;
import org.junit.Test;

import static org.hamcrest.CoreMatchers.isA;
import static org.junit.Assert.assertNotNull;

public class ClosePoolTest extends IndyIntegrationTest {

	@Test
	public void testClosePoolWorks() throws Exception {
		Pool pool = PoolUtils.createAndOpenPoolLedger();
		assertNotNull(pool);
		openedPools.add(pool);

		pool.closePoolLedger().get();
		openedPools.remove(pool);
	}

	@Test
	public void testClosePoolWorksForTwice() throws Exception {
		thrown.expectCause(isA(InvalidPoolException.class));

		Pool pool = PoolUtils.createAndOpenPoolLedger();
		assertNotNull(pool);
		openedPools.add(pool);

		pool.closePoolLedger().get();
		openedPools.remove(pool);
		pool.closePoolLedger().get();
	}

	@Test
	public void testClosePoolWorksForReopenAfterClose() throws Exception {
		String poolName = PoolUtils.createPoolLedgerConfig();

		Pool pool = Pool.openPoolLedger(poolName, null).get();
		assertNotNull(pool);

		pool.closePoolLedger().get();

		pool = Pool.openPoolLedger(poolName, null).get();
		openedPools.add(pool);
	}

	@Test
	public void testAutoCloseWorks() throws Exception {
		String poolName = PoolUtils.createPoolLedgerConfig();
		try (Pool pool = Pool.openPoolLedger(poolName, null).get()) {
			assertNotNull(pool);
		}
		Pool pool = Pool.openPoolLedger(poolName, null).get();
		openedPools.add(pool);
	}
}

File: wrappers/dotnet/indy-sdk-dotnet-test/PoolTests/ClosePoolTest.cs
using Hyperledger.Indy.PoolApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.PoolTests
{
    [TestClass]
    public class ClosePoolTest : IndyIntegrationTestBase
    {
        [TestMethod]
        public async Task TestClosePoolWorks()
        {
            var pool = await PoolUtils.CreateAndOpenPoolLedgerAsync();
            Assert.IsNotNull(pool);
            openedPools.Add(pool);

            await pool.CloseAsync();
            openedPools.Remove(pool);
        }

        [TestMethod]
        public async Task TestClosePoolWorksForTwice()
        {
            var pool = await PoolUtils.CreateAndOpenPoolLedgerAsync();
            Assert.IsNotNull(pool);
            openedPools.Add(pool);

            await pool.CloseAsync();
            openedPools.Remove(pool);

            var ex = await Assert.ThrowsExceptionAsync<InvalidPoolException>(() =>
                pool.CloseAsync()
            );
        }

        [TestMethod]
        public async Task TestClosePoolWorksForReopenAfterClose()
        {
            var poolName = PoolUtils.CreatePoolLedgerConfig();
            var pool = await Pool.OpenPoolLedgerAsync(poolName, null);

            Assert.IsNotNull(pool);

            await pool.CloseAsync();

            pool = await Pool.OpenPoolLedgerAsync(poolName, null);
            openedPools.Add(pool);
        }

        [TestMethod]
        public async Task TestAutoCloseWorks()
        {
            var poolName = PoolUtils.CreatePoolLedgerConfig();

            using (var autoClosePool = await Pool.OpenPoolLedgerAsync(poolName, null))
            {
                Assert.IsNotNull(autoClosePool);
            }

            var pool = await Pool.OpenPoolLedgerAsync(poolName, null);
            openedPools.Add(pool);
        }
    }
}
